WordPress/src/main/java/org/wordpress/android/ui/DeepLinkingIntentReceiverViewModel.kt

```diff
@@ -25,30 +25,35 @@ class DeepLinkingIntentReceiverViewModel
     val navigateAction = _navigateAction as LiveData<Event<NavigateAction>>
     val toast = editorLinkHandler.toast
 
+    /**
+     * Handles the following URLs
+     * `wordpress.com/post...`
+     * `wordpress.com/stats...`
+     * `public-api.wordpress.com/mbar`
+     * and builds the navigation action based on them
+     */
+    fun handleUrl(uriWrapper: UriWrapper): Boolean {
+        return if (shouldHandleUrl(uriWrapper)) {
+            _navigateAction.value = Event(buildNavigateAction(uriWrapper))
+            true
+        } else {
+            false
+        }
+    }
+
     /**
      * This viewmodel handles the following URLs
      * `wordpress.com/post/...`
      * `wordpress.com/stats/...`
      * `public-api.wordpress.com/mbar/`
      * In these cases this function returns true
      */
-    fun shouldHandleUrl(uri: UriWrapper): Boolean {
+    private fun shouldHandleUrl(uri: UriWrapper): Boolean {
         return isTrackingUrl(uri) ||
                 editorLinkHandler.isEditorUrl(uri) ||
                 statsLinkHandler.isStatsUrl(uri)
     }
```

Comment:
This is definitely an improvement but I was wondering if it wouldn't be possible to completely remove this method now. I tinkered a bit with this idea and I believe it's possible, but it could potentially change some implicit behaviors, so I'll leave it for another PR.